2/step3.py

```diff
@@ -0,0 +1,18 @@
+class Solution:
+    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
+        carry = 0
+        head_node = ListNode()
+        tail_node = head_node
+        while l1 is not None or l2 is not None or carry == 1:
+            l1_digit = l1.val if l1 is not None else 0
+            l2_digit = l2.val if l2 is not None else 0
+            total = l1_digit + l2_digit + carry
+            node = ListNode(total % 10)
+            tail_node.next = node
```

Comment:
```suggestion
            tail_node.next = ListNode(total % 10)
```
としてしまってnodeという変数を消してもよいかなと思いました